main.go

```diff
@@ -63,181 +51,335 @@ func init() {
+	flags.Float32Var(&config.TopP, "topp", config.TopP, stdoutStyles.FlagDesc.Render(help["topp"]))
+	flags.UintVar(&config.Fanciness, "fanciness", config.Fanciness, stdoutStyles.FlagDesc.Render(help["fanciness"]))
+	flags.StringVar(&config.StatusText, "status-text", config.StatusText, stdoutStyles.FlagDesc.Render(help["status-text"]))
+	flags.BoolVar(&config.ResetSettings, "reset-settings", config.ResetSettings, stdoutStyles.FlagDesc.Render(help["reset-settings"]))
+	flags.StringVarP(&config.Title, "title", "t", config.Title, stdoutStyles.FlagDesc.Render(help["title"]))
+	flags.StringVar(&config.Delete, "delete", config.Delete, stdoutStyles.FlagDesc.Render(help["delete"]))
+	flags.StringVarP(&config.Show, "show", "s", config.Show, stdoutStyles.FlagDesc.Render(help["show"]))
+
+	for _, name := range []string{"show", "delete", "continue"} {
+		_ = rootCmd.RegisterFlagCompletionFunc(name, func(_ *cobra.Command, _ []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			results, _ := db.Completions(toComplete)
+			return results, cobra.ShellCompDirectiveDefault
+		})
 	}
 
-	if mods.Config.ResetSettings {
-		resetSettings(mods)
+	flags.BoolVar(&config.NoCache, "no-cache", config.NoCache, stdoutStyles.FlagDesc.Render(help["no-cache"]))
+	flags.Lookup("prompt").NoOptDefVal = "-1"
+	flags.SortFlags = false
+	// flags.Init("", flag.ContinueOnError)
+
+	if config.Format && config.FormatText == "" {
+		config.FormatText = "Format the response as markdown without enclosing backticks."
 	}
+}
 
-	if mods.Config.ShowHelp || (mods.Input == "" &&
-		mods.Config.Prefix == "" &&
-		mods.Config.Show == "" &&
-		mods.Config.Delete == "" &&
-		!mods.Config.List) {
-		flag.Usage()
-		exit(mods, 0)
+func main() {
+	var err error
+	config, err = newConfig()
+	if err != nil {
+		handleError(modsError{err, "Could not load your configuration file."})
+		os.Exit(1)
 	}
 
-	if mods.Config.Show != "" {
-		exit(mods, 0)
+	cache = newCache(config.CachePath)
+	db, err = openDB(filepath.Join(config.CachePath, "mods.db"))
+	if err != nil {
+		handleError(modsError{err, "Could not open database."})
+		os.Exit(1)
 	}
+	defer db.Close() //nolint:errcheck
+
+	// XXX: this must come after creating the config.
+	initFlags()
 
-	if mods.Config.List {
-		listConversations(mods)
+	// XXX: since mods doesn't have any subcommands, Cobra won't create the
+	// default `completion` command.
+	// Forcefully create the completion related subcommands by adding a fake
+	// command when completions are being used.
+	if os.Getenv("__MODS_CMP_ENABLED") == "1" || (len(os.Args) > 1 && os.Args[1] == "__complete") {
+		rootCmd.AddCommand(&cobra.Command{Use: "____fake_command_to_enable_completions"})
+		rootCmd.InitDefaultCompletionCmd()
 	}
 
-	if mods.Config.Delete != "" {
-		deleteConversation(mods)
+	if err := rootCmd.Execute(); err != nil {
+		handleError(err)
+		_ = db.Close()
```

Comment:
might not need this as we close the db in a defer in L217

Reply:
os.Exit in the next line will ignore defers, so this is needed yes